test/infamy/furl.py

```diff
@@ -0,0 +1,25 @@
+"""Fugly URL fetcher"""
+
+import urllib.error
+import urllib.request
+
+class Furl:
+    """Furl wraps urllib in a way similar to curl"""
+
+    def encode(self, string):
+        """URL encode a string"""
+        return urllib.parse.quote(string, safe='/:')
```

Comment:
```suggestion
    def __init__(self, string):
        """Create a new URL checker"""
        self.url = urllib.parse.quote(string, safe='/:')
```
Looking at the use of `Furl`s in the tests, it looks like the URL could be an argument to a constructor instead. Then neither `check()` nor `nscheck()` would have to take a URL parameter, it would use the one this particular `Furl` is bound to (`self.url`).

So tests could do things like:
```python
the_goog = Furl("https://google.com")

assert(the_goog.check("I'm Feeling Lucky"))
```

Reply:
Nice one, fixing!